### PR TITLE
Remove remaining conversion's warnings.

### PR DESCRIPTION
**Pull request by drslebedev**
Continue removing remaining conversion's warnings after #79.

### CODE REVIEW

File: test/qa_buffer.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ const boost::ut::suite UserApiExamples = [] {
             auto lambda = [](auto w) { // test writer generating consecutive samples
                 static int offset = 1;
                 std::iota(w.begin(), w.end(), offset);
-                offset += w.size();
+                offset += static_cast<int>(w.size());
```
Comment (RalphSteinhagen) [inline test/qa_buffer.cpp:202]: N.B. `auto w` is actually `int32_t w` -> cast should probably be `static_cast<int32_t>(w.size());`

File: test/qa_buffer.cpp [hunk 2]
```diff
@@ -219,7 +219,7 @@ const boost::ut::suite UserApiExamples = [] {
             const auto demoWriter = [](auto w) {
                 static int offset = 1;
                 std::iota(w.begin(), w.end(), offset);
-                offset += w.size();
+                offset += static_cast<int>(w.size());
```
Comment (RalphSteinhagen) [inline test/qa_buffer.cpp:222]: same as a above ... maybe change the `auto w` to `int32_t w` to be more explicit/expressive for future readers?!?

same for the other occurrences further below.

File: test/qa_dynamic_port.cpp [hunk 3]
```diff
@@ -125,17 +129,17 @@ const boost::ut::suite PortApiTests = [] {
         auto lambda = [&offset](auto &w) {
```
Comment (RalphSteinhagen) [inline test/qa_dynamic_port.cpp:129]: same here ... auto == int32_t

File: test/qa_scheduler.cpp [hunk 4]
```diff
@@ -20,17 +20,17 @@ trace(trace_vector &traceVector, std::string_view id) {
 }
 
 // define some example graph nodes
-template<typename T, std::int64_t N>
+template<typename T, std::make_signed_t<std::size_t> N>
```
Comment (RalphSteinhagen) [inline test/qa_scheduler.cpp:23]: should be std::size_t

File: test/qa_scheduler.cpp [hunk 5]
```diff
@@ -20,17 +20,17 @@ trace(trace_vector &traceVector, std::string_view id) {
...
 class count_source : public fg::node<count_source<T, N>, fg::OUT<T, 0, std::numeric_limits<std::size_t>::max(), "out">> {
-    trace_vector &tracer;
-    std::int64_t  count = 0;
+    trace_vector                   &tracer;
+    std::make_signed_t<std::size_t> count = 0;
```
Comment (RalphSteinhagen) [inline test/qa_scheduler.cpp:26]: should be std::size_t

File: test/qa_settings.cpp [hunk 6]
```diff
@@ -85,7 +85,7 @@ struct Source : public node<Source<T>> {
     void
     init(const tag_t::map_type &old_settings, const tag_t::map_type &new_settings) {
         // optional init function that is called after construction and whenever settings change
-        fair::graph::publish_tag(out, { { "n_samples_max", n_samples_max } }, n_tag_offset);
+        fair::graph::publish_tag(out, { { "n_samples_max", n_samples_max } }, static_cast<std::size_t>(n_tag_offset));
```
Comment (RalphSteinhagen) [inline test/qa_settings.cpp:88]: n_tag_offset should probably be std::size_t

File: test/qa_settings.cpp [hunk 7]
```diff
@@ -166,7 +166,7 @@ struct Sink : public node<Sink<T>> {
         */
 
         if constexpr (fair::meta::any_simd<V>) {
-            n_samples_consumed += V::size();
+            n_samples_consumed += static_cast<std::int32_t>(V::size());
```
Comment (RalphSteinhagen) [inline test/qa_settings.cpp:169]: n_samples_consumed should probably be std::size_t

**Review by RalphSteinhagen — CHANGES_REQUESTED** [review]
Thanks for the changes... many conversion warnings are gone. 

While the castings remove the compiler warning, this also shadows an underlying design deficiency that should be fixed at the source. For some changes, I think it would be better to avoid `static_cast<>` by changing the variable type in the first place.

I noted some of the places where I feel this would improve the overall design so that we do not have to touch the code twice.

N.B. this PR correctly fixes the `-Wconversion` warnings but we should eventually also look into the other ones. We should all start looking a bit deeper/closer to those as well.

File: test/qa_buffer.cpp (hunk 1 above)
Comment (drslebedev) [reply test/qa_buffer.cpp:202]: I think here everything is correct. 'w' is 'std::span<int>', but I cast 'w.size()' to be the same type as offset, which is 'int'. One of the possibility is to declare 'offset' as 'std::size_t'.

Comment (RalphSteinhagen) [reply test/qa_buffer.cpp:202]: Maybe 'std::size_t' would be easier and lowering the cognitive complexity ...

**Review by RalphSteinhagen — APPROVED** [review]
@drslebedev very good ... thanks for the refactoring, checking and eliminating the warnings! 

Job well done! :+1:

**Pull request merged**